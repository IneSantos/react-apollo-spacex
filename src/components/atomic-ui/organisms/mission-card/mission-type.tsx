export type Mission = {
    name: string,
    details: string,
    link: string,
    img: string
}