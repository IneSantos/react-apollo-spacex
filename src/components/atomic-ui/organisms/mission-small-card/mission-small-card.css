@import "../../../../theme-variables.css";

.mission-wrapper {
    background-color: var(--background-color);
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    margin-left: auto;
    margin-right: auto;

    padding: 1.5rem 2rem;
    margin: 1.3rem;
    border-radius: 5px;
    position: relative;
    -webkit-box-shadow: 0 4px 6px 4px rgb(91 164 164 / 20%);
            box-shadow: 0 4px 6px 4px rgb(91 164 164 / 20%);
    
    color: #ccc;
    border: 3px solid var(--card-border-color);

    background: -webkit-gradient(linear, right top, left top, color-stop(50%, var(--background-color)), color-stop(50%, var(--text-color))) right;

    background: -o-linear-gradient(right, var(--background-color) 50%, var(--text-color) 50%) right;

    background: linear-gradient(to left, var(--background-color) 50%, var(--text-color) 50%) right;
    background-size: 200%;
    -webkit-transition: .5s ease-out;
    -o-transition: .5s ease-out;
    transition: .5s ease-out;
}

.mission-wrapper:hover { 
    background-position: left;
    background-color: var(--text-color);
    color: var(--background-color);
    -webkit-box-shadow: 0 1px 3px 0 rgb(60 64 67 / 30%), 0 4px 8px 3px rgb(60 64 67 / 15%);
    box-shadow: 0 1px 3px 0 rgb(60 64 67 / 30%), 0 4px 8px 3px rgb(60 64 67 / 15%);
}

.mission-wrapper:hover > .card-header {
    color: var(--background-color);
}

.mission-wrapper:hover > .list-card-body {
    color: var(--background-color);
}

.mission-wrapper:first-child {
    margin: 1.3rem;
}

@media (min-width: 60em) {
    .mission-wrapper {
        border-radius: 3px;
        border: 1px solid var(--card-border-color);
        -webkit-box-shadow: none;
        box-shadow: none;
        font-size: 16px;
        line-height: 28px;
        max-width: 980px;
        margin: 1.3rem 0;
    }

    .mission-wrapper:first-child {
        margin: 1.3rem 0;
    }
}