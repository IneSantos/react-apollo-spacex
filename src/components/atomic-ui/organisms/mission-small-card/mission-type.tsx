export type Mission = {
    id: string,
    name: string,
    details: string,
    date: string
}