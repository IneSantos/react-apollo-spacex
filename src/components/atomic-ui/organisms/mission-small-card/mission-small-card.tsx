import { Link } from "react-router-dom";
import ListCardBody from "../../molecules/list-card-body/list-card-body";
import CardHeader from "../../molecules/card-header/card-header";
import { Mission } from "./mission-type";
import "./mission-small-card.css";

const MissionSmallCard = (props: Mission) => {
  const { id, name, details, date } = props;
  return (
    <Link to={`/mission/${id}`}  style={{ textDecoration: "none" }}>
      <div className="mission-wrapper">
        <CardHeader name={name} />
        <ListCardBody details={details} date={date} />
      </div>
    </Link>
  );
}

export default MissionSmallCard;