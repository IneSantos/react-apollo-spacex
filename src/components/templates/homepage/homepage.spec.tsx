import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Mission } from "../../atomic-ui/organisms/mission-small-card/mission-type";
import HomepageTemplate from "./homepage";

describe("HomepageTemplate template tests", () => {
    test(`HomepageTemplate component render with no missions`, () => {
        const { container } = render(<HomepageTemplate missions={[]}/>);
        expect(container.firstChild).toMatchSnapshot();
    });

    test(`HomepageTemplate component render with missions`, () => {
        const missions: Mission[] = [
            {
                id: "0",
                name: "This is a mission name",
                details: "This is a mission detail",
                date: "2022/04/10"
            },
            {
                id: "1",
                name: "This is another mission name",
                details: "This is another mission detail",
                date: "2022/04/11"
            }
        ];

        const { container } = render(
            <Router>
                  <HomepageTemplate missions={missions}/>
            </Router>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
