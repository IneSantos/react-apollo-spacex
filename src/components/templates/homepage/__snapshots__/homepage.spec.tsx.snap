// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageTemplate template tests HomepageTemplate component render with missions 1`] = `
<div
  class="spacex-missions"
>
  <header>
    <div
      class="banner"
    />
    <h1>
      Recent Launches
    </h1>
  </header>
  <section
    class="missions-body"
  >
    <div
      class="mission-list"
    >
      <a
        href="/mission/0"
        style="text-decoration: none;"
      >
        <div
          class="mission-wrapper"
        >
          <div
            class="card-header"
          >
            <h2
              class="title"
            >
              This is a mission name
            </h2>
          </div>
          <div
            class="list-card-body"
          >
            <div
              class="details"
            >
              This is a mission detail
            </div>
            <div
              class="launch-date"
            >
              April 10, 2022
            </div>
          </div>
        </div>
      </a>
      <a
        href="/mission/1"
        style="text-decoration: none;"
      >
        <div
          class="mission-wrapper"
        >
          <div
            class="card-header"
          >
            <h2
              class="title"
            >
              This is another mission name
            </h2>
          </div>
          <div
            class="list-card-body"
          >
            <div
              class="details"
            >
              This is another mission detail
            </div>
            <div
              class="launch-date"
            >
              April 11, 2022
            </div>
          </div>
        </div>
      </a>
    </div>
  </section>
</div>
`;

exports[`HomepageTemplate template tests HomepageTemplate component render with no missions 1`] = `
<div
  class="spacex-missions"
>
  <header>
    <div
      class="banner"
    />
    <h1>
      Recent Launches
    </h1>
  </header>
  <section
    class="missions-body"
  >
    <div
      class="mission-list"
    >
      <div
        class="empty-list"
      >
        <h3>
          No results.
        </h3>
        <span>
          Sorry we have no SpaceX Missions to show.
        </span>
      </div>
    </div>
  </section>
</div>
`;
